allennlp_rc/models/transformer_qa.py

```diff
@@ -0,0 +1,233 @@
+import logging
+from typing import Any, Dict, List, Optional
+
+import numpy as np
+import torch
+from allennlp.common.util import sanitize_wordpiece
+from torch import nn
+
+from allennlp.data import Vocabulary
+from allennlp.models.model import Model
+from allennlp.modules import TextFieldEmbedder
+from allennlp.nn import util, RegularizerApplicator
+from allennlp.training.metrics import BooleanAccuracy, CategoricalAccuracy
+from torch.nn.functional import cross_entropy
+
+from allennlp_rc.models.util import get_best_span
+from allennlp_rc.eval import SquadEmAndF1
+
+logger = logging.getLogger(__name__)
+
+
+@Model.register("transformer_qa")
+class TransformerQA(Model):
+    """
+    This class implements a reading comprehension model patterned after the SQuAD model in the transformers
+    project.
+
+    It predicts start tokens and end tokens with a linear layer on top of the embeddings.
+
+    Parameters
+    ----------
+    vocab : ``Vocabulary``
+    text_field_embedder : ``TextFieldEmbedder``
+        Used to embed the ``question`` and ``passage`` ``TextFields`` we get as input to the model. These would
+        usually be ``pretrained_transformer`` embedders, but they don't have to be.
+    regularizer : ``RegularizerApplicator``, optional (default=``None``)
+        If provided, will be used to calculate the regularization penalty during training.
+    """
+
+    def __init__(
+        self,
+        vocab: Vocabulary,
+        text_field_embedder: TextFieldEmbedder,
+        regularizer: Optional[RegularizerApplicator] = None,
+    ) -> None:
+        super().__init__(vocab, regularizer)
+        self._linear_layer = nn.Linear(text_field_embedder.get_output_dim(), 2)
+        self._text_field_embedder = text_field_embedder
+
+        self._span_start_accuracy = CategoricalAccuracy()
+        self._span_end_accuracy = CategoricalAccuracy()
+        self._span_accuracy = BooleanAccuracy()
+        self._metrics = SquadEmAndF1()
```

Comment:
This name is a bit confusing.

Reply:
It's now called `_per_instance_metrics`, from a previous comment.

Reply:
Aren't the other metrics "per instance" also?

Reply:
The official metrics are "per question".